### DISCUSSION TITLE
How to work with webinspector?

### DISCUSSION THREAD

**Discussion by icyflatline** [doronz88/pymobiledevice3#450] [Q&A]
Hi,
Is it possible to get some explanation of how to work with `webinspector ` in `pymobiledevice3`?

What does `cdp` do? I mean, I get that it runs some kind of server on 9222 port by default, but what can I do with it?
What kind of things are possible to do in` js-shell`? How to exit` js-shell`?
Can I debug opened tabs in iOS Safari on a device somehow? Like, get HTML source/run console commands/find elements/etc?

Thanks in advance.

**Comment by doronz88** [comment]
Hi! I created a PR that tries to explain these interfaces a bit more. Please look at: #451 and tell me your opinion.
This should provide answers for:
- Interacting with the CDP server (using Google Chrome inspector)
- What `js-shell` provides
  - Exiting is just like any other shell, be sending a `EOF` (`Ctrl-D`)
